src/value/reader.rs

```diff
@@ -257,30 +253,16 @@ mod reader_tests {
             }
         "#,
         vec![
-            vec![
-                (Symbol::owned("string_field"), String("oink!".into())),
-                (Symbol::owned("string_field"), String("moo!".into())),
-                (Symbol::owned("bool_field"), Boolean(true)),
-            ]
-        ].into_iter()
-            .map(|fields: Vec<(Symbol, Value)>| {
-                Struct(
-                    fields.into_iter().map(|(tok, val)| {
-                        (tok, Element::new(vec![Symbol::owned("a")], val))
-                    }).collect()).into()
-            })
-            .collect(),
+            ion_struct! {
+                "string_field": (["a"], "oink!"),
+                "string_field": (["a"], "moo!"),
+                "bool_field": (["a"], true)
```

Comment:
🗺️ `Into<Element>` is implemented for tuples of `([...annotations...], value)`, allowing for this syntax. I'm interested in getting feedback on this; alternative ideas welcome.

I considered an `annotated!` method (`"bool_field": annotated!("a", true)`) but found it to be too verbose.

Reply:
Alternately, we could create a trait that extends `Into<Element>` with a function `with_annotations()`.

E.g.:
```
ion_struct! {
  a: 1.with_annotations(["foo"])
}
```

Reply:
> Alternately, we could create a trait that extends `Into<Element>` with a function `with_annotations()`.
> 
> E.g.:
> 
> ```
> ion_struct! {
>   a: 1.with_annotations(["foo"])
> }
> ```

_Whoa,_ now THAT's thinking with portals! I'm going to give this one a try, I really like it.